packages/nextjs/pages/api/etl/index.ts

```diff
@@ -99,11 +105,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       const day90 = dates[90];
       // Iterate over all projects
       for await (const project of Project.find({}).cursor()) {
```

Comment:
We could also build all the metric summary data in parallel to significantly speed things up

```js
      const projectMetricSummaryOps = [];

      const [projects, scoresToday, scoresDay7, scoresDay30, scoresDay90] = await Promise.all([
        Project.find({}).lean(),
        TempProjectScore.find({ date: today }).lean(),
        TempProjectScore.find({ date: day7 }).lean(),
        TempProjectScore.find({ date: day30 }).lean(),
        TempProjectScore.find({ date: day90 }).lean(),
      ]);
      // Fetch all required scores at once

      // Create a map for quick lookup
      const createScoreMap = scores => {
        return scores.reduce((acc, score) => {
          if (!acc[score.projectId]) {
            acc[score.projectId] = {};
          }
          acc[score.projectId][score.metricName] = score;
          return acc;
        }, {});
      };

      const scoreMapToday = createScoreMap(scoresToday);
      const scoreMapDay7 = createScoreMap(scoresDay7);
      const scoreMapDay30 = createScoreMap(scoresDay30);
      const scoreMapDay90 = createScoreMap(scoresDay90);

      projects.forEach(project => {
        console.log(`Processing project ${project.name}`);
        const projectId = project.id;

        for (const metric of Object.keys(metricNamesObj) as (keyof Metrics)[]) {
          const scoreToday = scoreMapToday[projectId]?.[metric] || 0;
          const scoreDay7 = scoreMapDay7[projectId]?.[metric] || 0;
          const scoreDay30 = scoreMapDay30[projectId]?.[metric] || 0;
          const scoreDay90 = scoreMapDay90[projectId]?.[metric] || 0;

          const metricSummary = {
            date: today,
            projectId,
            metricName: metric,
            7: getMovement(scoreToday, scoreDay7),
            30: getMovement(scoreToday, scoreDay30),
            90: getMovement(scoreToday, scoreDay90),
          };

          projectMetricSummaryOps.push({
            insertOne: {
              document: metricSummary,
            },
          });
        }
      });

      // Batch insert project metric summaries
      if (projectMetricSummaryOps.length > 0) {
        await TempProjectMetricSummary.bulkWrite(projectMetricSummaryOps);
      }
```